### DISCUSSION TITLE
Want to fire an event on server to all client (need to communicate a bool value to all clients) before calling initialise()

### DISCUSSION THREAD

**Discussion by hemant-works** [NVIDIA/NVFlare#737] [Q&A] [answered]
I wanted to know if there is any mechanism for the Scatter and Gather workflow to pass information (boolean value in my case) to all the clients, before the clients run their respective initialise() function.

**Comment by holgerroth** [comment]
Thanks for your question. What initialize function are you referring to?

**Reply by hemant-works** [reply]
On the client side, I have a custom components which inherits the **Learner** class. This custom component has three functions : 1. train(), 2. submit_model(), 3. validate()
Basically this is the python script that a client should run. It also has an initialize() function, which intializes all the initial settings on the client side.

**Comment by holgerroth** [comment, marked as answer]
I see. Please consider if the flag information is really needed in the initialize() function. You could try sending your flag as a property of the [data shareable](https://github.com/NVIDIA/NVFlare/blob/38c2b32396acc609eb13f21a148b3f4fa18fbda2/nvflare/app_common/workflows/scatter_and_gather.py#L191) for each task when it's needed. Otherwise, there is a timing issue as nvflare is a distributed system. All FL components you specify in your configs might get initialized at different times on server and clients. The Learner on the clients might get initialized() even before the flag information is available on the server.

**Reply by hemant-works** [reply]
Okay, I needed to know that. Thanks for your time.
